src/main/java/core/operations/BalanceOperation.java

```
@@ -0,0 +1,15 @@
+package core.operations;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.StorageDao;
+import core.interfaces.OperationHandler;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction tx, StorageDao storageDao) {
+        if (tx != null && tx.getQuantity() >= 0) {

```

The code currently ignores null `tx` or negative quantities by returning silently. The checklist requires validating inputs (null parameters, negative quantity). Instead of skipping, validate and throw a descriptive RuntimeException that includes context (operation, fruit, quantity or the raw line) so failures are easy to debug. Fail-fast behavior is required.